src/encode.rs

```diff
@@ -42,8 +62,17 @@ impl<'a, I: AsRef<[u8]>> EncodeBuilder<'a, I> {
     /// ```
     pub fn into_string(self) -> String {
         let input = self.input.as_ref();
-        let mut output = String::with_capacity((input.len() / 5 + 1) * 8);
-        encode_into(input, &mut output, self.alpha);
+
+        let checksum_capacity = match self.check {
+            true => CHECKSUM_LEN,
+            false => 0
+        };
+
+        let mut output = String::with_capacity((input.len() / 5 + 1) * 8 + checksum_capacity);
```

Comment:
I think this should be `String::with_capacity(((input.len() + checksum_capacity) / 5 + 1) * 8)`, the checksum length is the number of decoded bytes, so might not need an entire 4 bytes extra in the output depending on the input length.

Reply:
good point, I will change

Reply:
Done